examples/systemcalls/systemcalls.c

```diff
@@ -17,6 +26,30 @@ bool do_system(const char *cmd)
  *   or false() if it returned a failure
 */
 
+    /* check for null cmd */
+    if(cmd == NULL)
+    {
+        printf("\r\ncmd arguement invalid.");
+        return false;
+    }
+    
+    /* attempt to create new process with given command */
+    int result = system(cmd);
+
+    /* error handling for system call failures */
+    if(result == -1)
+    {
+        printf("\r\nsystem call returned -1, child process could not be created.");
+        return false;
+    }
+    else if(result == 127)
```

Comment:
Is 1 and 127 the only error code which should result in a false return code?  I think there may be others you need to handle here.